docs/architecture/decisions/implement-cli-framework_001.md

```diff
@@ -0,0 +1,58 @@
+
+This ADR also outlines the adoption of environment variables and a configuration file within the CLI.  These will provides alternatives methods of passing arguments to the CLI beyond just command flags.  This provides users with flexibility in how they pass arguments to the CLI and creates a more static option for arguments that tend to remain unchanged between command executions.
+
+
+## Decision
+
+The trestlebot module will be refactored to remove the use of `argparse` in favor of Click as the CLI framework.  The code contained in `entrypoints` will be converted into Click commands under the `trestlebot` CLI application.  A new `cli.py` module will be created as the main entrypoint.
+
+In addition, support will be added for using a configuration file and environment variables as CLI inputs.  The CLI will prioritize arguments passed as command flags.  If no argument is passed, the CLI will check for an environment variable.  Finally, if no enviroment variable is found, it will look to the configuration file.  
+
+The configuration file will be broken into two primary categories, `global` and `model specific`.  Global configuration will apply across all models and include values such as git provider, markdown directories, etc.  Model specific configuration will apply to the given OSCAL model only.  While it is expected that most repos will be used for authoring a single OSCAL model, the possiblity of authoring more than one model would be supported.  The configuration file would be initialized at a default location during the `trestlebot init` command.  Manual creation and editing is also possible.  The path to the configuration file can be passed using the `--config | -c` flag.  This would not be required if using the default file location. 
+
+- the default configuration file location will be `.trestlebot/config.yaml`
+- all environment variables will have a `TRESTLEBOT_` prefix
+
+#### Example config:
+
+```yaml
+---
+version: 1
+working-dir: "."
+markdown-path-ssp: markdown/system-security-plans
+markdown-path-profile: markdown/proflies
+markdown-path-catalog: markdown/catalogs
+markdown-path-compdef: markdown/component-definitions
+ssp-index-path: ssp-index.json
+profile-upstreams: [<url>]
+catalog-upstreams: [<url>]
+compdef-upstreams: [<url>]
+git-provider-type: github
+git-provider-url: github.com
+git-committer-name: "Foo Bar"
+git-committer-email: foo@bar.com
+models:
+  # we could allow for multiple or keep this as one
+  - oscal-model: "ssp"
```

Comment:
This configuration would be per model type so I think `markdown-path` makes more sense here. As well as `ssp-index`. My line of thinking is if you need to specify a model type on the field, it should go under the model specific configuration. WDYT?

Reply:
Yes, that makes sense.  Are there cases where more than one markdown path would be needed for a given model?  For example, using autosync on a repo that has both "compdef" and "ssp" model authoring?  Both may need to know the "compdef" markdown path, correct?

Reply:
I think that could be use case and I thought the config might look like this for that use case:

```
version: 1
working-dir: "."
git-provider-type: github
git-provider-url: github.com
git-committer-name: "Foo Bar"
git-committer-email: foo@bar.com
models:
  - oscal-model: "ssp"
    markdown-path: markdown/system-security-plans
    ssp-index-path: ssp-index.json
  - oscal-model: "compdef"
    markdown-path: markdown/component-definitions
  # ... other models ...
 ```
 
 WDYT?

Reply:
Ok, model specific makes sense.